Tests/CoreTests.swift

```diff
@@ -126,12 +142,148 @@ class CoreTests: QuickSpec {
                 }
             }
 
+            context("Destroy") {
+                it("Should trigger willDestroy") {
+                    var didCallWillDestroy = false
+
+                    core.on(InternalEvent.willDestroy.rawValue) { _ in
+                        didCallWillDestroy = true
+                    }
+                    core.destroy()
+                    
+                    expect(didCallWillDestroy).toEventually(beTrue())
+                }
+
+                it("Should trigger didDestroy") {
+                    var didCallDidDestroy = false
+
+                    core.on(InternalEvent.willDestroy.rawValue) { _ in
+                        didCallDidDestroy = true
+                    }
+                    core.destroy()
+
+                    expect(didCallDidDestroy).toEventually(beTrue())
+                }
+
+                it("Should remove listeners") {
```

Comment:
This test is doing more than what the description informs: it checks for the correctness of the `listenTo` and then check if the listener was removed. We could break this for better legibility.